### DISCUSSION TITLE
failure during install

### DISCUSSION THREAD

**Discussion by neverdimed** [andmarti1424/sc-im#712] [General]
i asked my Q on askubuntu and didnt get a response, i hope someone here can help (https://askubuntu.com/questions/1403027/sc-im-failure-during-install):

using: ubuntu 18.04 and git version 2.17.1

when i try to install SC-IM as per the instructions for ubuntu: https://github.com/andmarti1424/sc-im/wiki/Ubuntu-with-XLSX-import-&-export

i see 'undefined reference' when i run make, as follows:

[...]/sc-im/src/tui.c:1535: undefined reference to `init_extended_pair'
color.o: In function `redefine_color':
[...]/sc-im/src/color.c:594: undefined reference to `init_extended_color'
color.o: In function `define_color':
[...]/sc-im/src/color.c:661: undefined reference to `init_extended_color'
collect2: error: ld returned 1 exit status
Makefile:194: recipe for target 'sc-im' failed
make: *** [sc-im] Error 1

then when i run sudo make install i see:

install -d /usr/local/bin
install sc-im /usr/local/bin/sc-im
install: cannot stat 'sc-im': No such file or directory
Makefile:178: recipe for target 'install' failed
make: *** [install] Error 1

they say here: https://github.com/andmarti1424/sc-im/issues/253 "try linking against ncursesw", and then: "That solved it." But what does this mean? I googled and didnt find anything useful. Thanks.

edit: Maybe this is relevant?: https://github.com/andmarti1424/sc-im/wiki/Ubuntu-16.04.1-with-libxlsreader I added these 3 lines to the Makefile in sc-im/src but it made no difference

thanks

**Comment by andmarti1424** [comment]
I dont use ubuntu. But did you check you have ncurses installed? ncurses-dev package may be?

**Reply by neverdimed** [reply]
actually i already have them installed, as per: https://www.cyberciti.biz/faq/linux-install-ncurses-library-headers-on-debian-ubuntu-centos-fedora/

what does the person mean here when they say "try linking against ncursesw"? https://github.com/andmarti1424/sc-im/issues/253

cheers

**Reply by andmarti1424** [reply]
It would probably be better explained in here: https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html

**Reply by neverdimed** [reply]
sorry, i have no idea what that link is supposed to be telling me

i followed this very helpful video: "compile a text-based spreadsheet program - sc-im in Ubuntu 18.04 LTS " https://asciinema.org/a/357755 But there were still errors referring to define_color and redefine_color. I commented out the relevant bits of code in the relevant files and finally sc-im installed, when i run it there are no colours obviously and i cant see anything, so not a great solution

any help? cheers

**Comment by neverdimed** [comment]
i found the solution, it relates to this discussion: https://github.com/andmarti1424/sc-im/issues/482

ie, i changed init_extended_color to init_color in color.c (in two places) and init_extended_pair to init_pair in tui.c, then it installs fine. 

I have ncurses version 6.1, and as you say in the discussion "You might need version 6.2"

i look foreward to playing with sc-im now, thanks

**Reply by andmarti1424** [reply]
Be aware that you might not get rgb color definitions with ncurses 6.1

**Reply by neverdimed** [reply]
ok, it looks cool though, good enough until i can upgrade my system

**Reply by neverdimed** [reply]
i cant sponsor you thru github? i dont have patreon. Are github good to developers are they take a %?

**Reply by andmarti1424** [reply]
> i cant sponsor you thru github? i dont have patreon. Are github good to developers are they take a %?

They all take some % I'm afraid. Only PayPal or patreon by now.
